src/assets/TokensController.ts

```diff
@@ -505,66 +647,120 @@ export class TokensController extends BaseController<
    */
   removeAndIgnoreToken(address: string) {
     address = toChecksumHexAddress(address);
-    const { tokens, ignoredTokens } = this.state;
+    const { tokens, ignoredTokens, detectedTokens } = this.state;
 
     const alreadyIgnored = ignoredTokens.find(
       (tokenAddress) => tokenAddress.toLowerCase() === address.toLowerCase(),
     );
 
-    const newTokens = tokens.filter((token) => {
-      if (token.address.toLowerCase() === address.toLowerCase()) {
-        !alreadyIgnored && ignoredTokens.push(address);
-        return false;
-      }
-      return true;
-    });
+    if (alreadyIgnored) {
+      return;
+    }
 
-    const { newAllTokens, newAllIgnoredTokens } = this._getNewAllTokensState(
-      newTokens,
-      ignoredTokens,
+    const newTokens = tokens.filter(
+      (token) => token.address.toLowerCase() !== address.toLowerCase(),
+    );
+    const newDetectedTokens = detectedTokens.filter(
+      (token) => token.address.toLowerCase() !== address.toLowerCase(),
     );
+    ignoredTokens.push(address);
```

Comment:
This will also directly modify state.

Reply:
Updated to not directly mutate state. This was happening in a bunch of other places in this controller so I went ahead and fixed those as well.